spackbot/workers.py

```diff
@@ -41,23 +45,105 @@ def is_up_to_date(output):
     return "nothing to commit" in output
 
 
-def report_style_failure(job, connection, type, value, traceback):
+def post_failure_message(job, msg):
     """
     Get the api token from the job metadata, use it to post a comment on
     the PR containing the excepttion encountered and stack trace.
 
     """
-    user_msg = comments.get_style_error_message(type, value, traceback)
-
     token = None
     if "token" in job.meta:
         token = job.meta["token"]
 
     url = job.meta["post_comments_url"]
-    data = {"body": user_msg}
+    data = {"body": msg}
 
     helpers.synchronous_http_request(url, data=data, token=token)
-    logger.error(user_msg)
+    logger.error(msg)
+
+
+def report_style_failure(job, connection, type, value, traceback):
+    user_msg = comments.format_error_message(
+        "I encountered an error attempting to format style.", type, value, traceback)
+    post_failure_message(job, user_msg)
+
+
+def report_rebuild_failure(job, connection, type, value, traceback):
+    user_msg = comments.format_error_message(
+        "I encountered an error attempting to rebuild everything.", type, value, traceback)
+    post_failure_message(job, user_msg)
+
+
+async def run_pipeline_task(event):
+    """
+    Send an api request to gitlab telling it to run a pipeline on the
+    PR branch for the associated PR.  Depending on the job metadata
+    """
+    job = get_current_job()
+    token = job.meta["token"]
+    rebuild_everything = True if "rebuild_everything" in job.meta and job.meta["rebuild_everything"] is True else False
+
+    async with aiohttp.ClientSession() as session:
+        gh = gh_aiohttp.GitHubAPI(session, REQUESTER, oauth_token=token)
+
+        # Get the pull request number
+        pr_url = event.data["issue"]["pull_request"]["url"]
+        number = pr_url.split("/")[-1]
```

Comment:
Since we're using Python 3 we could do:

```suggestion
        *_, number = pr_url.split("/")
```

I dunno if it's that much clearer.

Reply:
I don't know either, but it's more modern I think.  Went with it.